### ISSUE TITLE
Is pull --parallel smart enough?

### ISSUE THREAD

**Issue by Jean85** [docker/compose#5677]
Hello there! This is more of an inquiry than a proper issue.

I'm currently using Docker compose to ease my Docker CI pipeline, and I'm leveraging the parallel pull feature to do that to. My question is: if I pull in parallel multiple images with various layers in common, do the pull notice that and do the smart move of pulling them only once?

Thanks!

**Comment by shin-** [comment]
Unfortunately, there is no such mechanism in place at the `docker-compose` level because there is no way for Compose to be aware of those shared layers before downloading them.

**Comment by Jean85** [comment]
This is because Docker Compose delegates the pull to Docker? Because it would be possible if it used the Registry API, but I can understand that the Docker client abstraction is nice to have/use...
